plugin/core/open.py

```diff
@@ -28,7 +30,7 @@ def open_file(window: sublime.Window, file_path: str, flags: int = 0, group: int
             return value[0]
 
     # Prepare a new promise to be resolved by a future on_load event (see the event listener in main.py)
-    def fullfill(resolve: ResolveFunc) -> None:
+    def fullfill(resolve: ResolveFunc[sublime.View]) -> None:
```

Comment:
This should be `ResolveFunc[Optional[sublime.View]]`

Reply:
It might be clearer to do that but `ResolveFunc` already ensures that `None` can also be passed:

```py
class ResolveFunc(Protocol[T]):
    def __call__(self, value: Union[T, 'Promise[T]'] = None) -> None:
        ...
```

That's also why I've used `Protocol` for it as it's not possible to represent an optional argument with `Callable`.

You can see here that the types are correct.
![Screenshot 2020-11-16 at 20 52 42](https://user-images.githubusercontent.com/153197/99301056-ad4c8c80-284d-11eb-93aa-220a918a3f31.png)

But I could try anyway to make it Optional explicitly.